solvers/noncvx_pro2.py

```diff
@@ -0,0 +1,71 @@
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from numpy.linalg import norm
+    import scipy.optimize as sciop
+    from scipy.sparse import issparse
+
+
+class Solver(BaseSolver):
+    name = "noncvx-pro2"
+
+    stopping_strategy = 'iteration'
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+
+    def skip(self, X, y, lmbd, fit_intercept):
+        if fit_intercept:
+            return True, f"{self.name} does not handle fit_intercept"
+        if issparse(X):
+            return True, f"{self.name} does not support sparse design matrices"
```

Comment:
Agreed, we can add it later.
```suggestion
        # XXX: make this solver work with sparse matrices.
        if issparse(X):
            return True, f"{self.name} does not support sparse design matrices"
```